pkgs/net-doc/net/scribblings/server.scrbl

```diff
@@ -0,0 +1,274 @@
+#lang scribble/doc
+
+@(require "common.rkt"
+          racket/runtime-path
+          scribble/example
+          (for-label net/server
+                     openssl
+                     racket/contract
+                     racket/tcp
+                     racket/list))
+
+@title[#:tag "server"]{General-purpose Server}
+
+@defmodule[net/server]{The @racketmodname[net/server] library provides
+support for running general-purpose networked servers.}
+
+@defproc[(start-server [host (or/c #f string?)]
+                       [port listen-port-number?]
+                       [handle (-> input-port? output-port? any)]
+                       [#:reuse? reuse? any/c #t]
+                       [#:max-allow-wait max-allow-wait exact-nonnegative-integer? 4]
+                       [#:max-concurrent max-concurrent
+                                         (or/c +inf.0 exact-positive-integer?)
```

Comment:
I wonder if we should accept `(or/c +inf.0 natural-number/c)`. It would admittedly be a bit silly to start a server that would never accept any connections, but maybe it would be useful for testing or something.